### PR TITLE
Add feature for setting content encoding

### PR DESCRIPTION
**Pull request by ribase**
As discussed in #13 here a pull request

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 build:
-  image: golang:1.5
+  image: golang:1.6
```
Comment (ericanderson) [inline .drone.yml:2]: Was this necessary?

File: .drone.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ publish:
     username: $$DOCKER_USER
     password: $$DOCKER_PASS
     email: $$DOCKER_EMAIL
-    repo: plugins/drone-s3-sync
+    repo: ribase/drone-s3-sync
```
Comment (ericanderson) [inline .drone.yml:20]: Shouldn't be changing the official repo path in your PR

File: aws.go [hunk 3]
```diff
@@ -114,12 +143,14 @@ func (a *AWS) Upload(local, remote string) error {
 			Key:         aws.String(remote),
 			Body:        file,
 			ContentType: aws.String(contentType),
+			ContentEncoding: aws.String(contentEncoding),
```
Comment (ericanderson) [inline aws.go:146]: What happens if contentEncoding isn't set?

File: buildPush.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+export GOBIN=$GOPATH/bin
+export PATH=$PATH:$GOBIN
+
+make deps build
+make deps docker
+docker build --rm=true -t ribase/drone-s3-sync .
+docker push ribase/drone-s3-sync
```
Comment (ericanderson) [inline buildPush.sh:9]: This seems to be a file specific to you, not the community at large?

File: .gitignore [hunk 5]
```diff
@@ -23,5 +23,4 @@ _testmain.go
 *.test
 *.prof
 
-coverage.out
-drone-s3-sync
```
Comment (ericanderson) [inline .gitignore:27]: Why this change?

File: aws.go [hunk 6]
```diff
@@ -82,6 +83,33 @@ func (a *AWS) Upload(local, remote string) error {
...
+      				".cgz": gzip
+      				".tgz": gzip
+	*/
+	var contentEncoding string
+	if a.vargs.ContentEncoding.IsString() {
+		contentEncoding = a.vargs.ContentEncoding.String()
+	} else if !a.vargs.ContentEncoding.IsEmpty() {
+		contentMap := a.vargs.ContentEncoding.Map()
+		for patternExt := range contentMap {
+			if patternExt == fileExt {
```
Comment (nlf) [inline aws.go:106]: we use glob matching everywhere else so it feels counterintuitive to have this only work as a direct extension match. i would vote for making this a glob match also.

File: vendor/github.com/aws/aws-sdk-go/LICENSE.txt [hunk 7]
```diff
@@ -1,202 +0,0 @@
-
```
Comment (ericanderson) [inline vendor/github.com/aws/aws-sdk-go/LICENSE.txt:1]: Why unvendor these?

**Comment by nlf** [comment]
aside from my own comment i'm going to +1 everything @ericanderson brought up also. you did a little more than just add a new feature. my suggestion would be to revert everything except your changes to `aws.go` (and the `Dockerfile`, i'm ok with upping the alpine base version)

**Comment by ericanderson** [comment]
ps, you cant change .drone.yml and have the build work

**Comment by tboerger** [comment]
If you revert all the unrelated changes and it's still backward compatible I'm fine with it. Maybe we should use govend including the prune flag to keep the vendoring small and simple

**Comment by tboerger** [comment]
After merging it I can update the drone.sec

File: buildPush.sh (hunk 4 above)
Comment (ribase) [inline buildPush.sh:9]: True, was just for my case.

File: vendor/github.com/aws/aws-sdk-go/LICENSE.txt (hunk 7 above)
Comment (ribase) [inline vendor/github.com/aws/aws-sdk-go/LICENSE.txt:1]: Mistake

File: .drone.yml (hunk 2 above)
Comment (ribase) [inline .drone.yml:20]: Ok, sorry, that was my first PR.

File: .drone.yml (hunk 1 above)
Comment (ribase) [inline .drone.yml:2]: I got issues under OSX with 1.5, tried 1.6, works like charm.

File: .drone.yml (hunk 2 above)
Comment (ericanderson) [inline .drone.yml:20]: No worries. I'm just pointing out things I see that would keep it from getting merged.

File: aws.go (hunk 3 above)
Comment (ribase) [inline aws.go:146]: Difficult, set by extension leads often to wrong enconding...
Let it empty is ugly...
Any ideas?

File: aws.go (hunk 6 above)
Comment (ribase) [inline aws.go:106]: Ok, i just copied from "contentType"

File: .drone.yml (hunk 1 above)
Comment (tboerger) [inline .drone.yml:2]: But this doesn't get built on OSX :)

File: aws.go (hunk 3 above)
Comment (ericanderson) [inline aws.go:146]: I'm afraid I don't understand. Will this be backwards compatible if contentEncoding is empty?

Comment (ribase) [inline aws.go:146]: If contentEncoding isnt set, nothing will happen to the rest of the code.

**Comment by tvooo** [comment]
+1
This would be incredibly useful!

Thanks!

**Comment by craigwl** [comment]
+1... this would make drone-s3-sync even better :)

**Comment by divercraig** [comment]
This would be so helpful right now, is it nearly ready to merge?

**Comment by nlf** [comment]
closed in favor of #20
